pyproject.toml

```diff
@@ -0,0 +1,4 @@
+#pyproject.toml
+[build-system]
+# Minimum requirements for the build system to execute.
+requires = ['setuptools>=45.0', 'wheel', 'cython>=0.29', 'oldest-supported-numpy', 'mpi4py==3.1.1']
```

Comment:
@sean-engelstad, can we replace this line in the `pyproject.toml` with the following:

```
requires = ['setuptools>=45.0', 'wheel', 'cython>=0.29', 'oldest-supported-numpy',
            # Build against an old version (3.1.1) of mpi4py for forward compatibility
            "mpi4py==3.1.1; python_version<'3.11'",
            # Python 3.11 requires 3.1.4+
            "mpi4py==3.1.4; python_version>='3.11'"]
```

This should fix the issues @kejacobson was seeing on Python 3.11 while retaining compatibility for older Python versions

Reply:
@timryanb I went ahead and implemented your suggested pyproject.toml require statement.